cmd/kubeadm/app/util/tokens_test.go

```diff
@@ -83,3 +84,87 @@ func newSecretsWithToken(token string) *kubeadmapi.Secrets {
 	s.GivenToken = token
 	return s
 }
+
+func TestGenerateToken(t *testing.T) {
+	var genTest = []struct {
+		s kubeadmapi.Secrets
+		l int
+		n int
+	}{
+		{kubeadmapi.Secrets{}, 2, 6},
+	}
+
+	for _, rt := range genTest {
+		GenerateToken(&rt.s)
+		givenToken := strings.Split(strings.ToLower(rt.s.GivenToken), ".")
+		if len(givenToken) != rt.l {
+			t.Errorf(
+				"failed GenerateToken:\n\texpected: %d\n\t  actual: %d",
+				rt.l,
+				len(givenToken),
+			)
+		}
+		if len(givenToken[0]) != rt.n {
+			t.Errorf(
+				"failed GenerateToken:\n\texpected: %d\n\t  actual: %d",
+				rt.l,
+				len(givenToken),
+			)
+		}
+	}
+}
+
+func TestUseGivenTokenIfValid(t *testing.T) {
+	var tokenTest = []struct {
+		s        kubeadmapi.Secrets
+		expected bool
+	}{
+		{kubeadmapi.Secrets{GivenToken: ""}, false},         // GivenToken == ""
+		{kubeadmapi.Secrets{GivenToken: "noperiod"}, false}, // not 2-part '.' format
+		{kubeadmapi.Secrets{GivenToken: "abcd.a"}, false},   // len(tokenID) != 6
+		{kubeadmapi.Secrets{GivenToken: "abcdef.a"}, true},
+	}
+
+	for _, rt := range tokenTest {
+		actual, _ := UseGivenTokenIfValid(&rt.s)
+		if actual != rt.expected {
+			t.Errorf(
+				"failed UseGivenTokenIfValid:\n\texpected: %t\n\t  actual: %t",
```

Comment:
Making this error more descriptive (pulling out the given token, for example) will make this easier to debug if it ever fails.